### PR TITLE
Update authentication API to conform to application API shape

### PR DESCRIPTION
**Pull request by ianyong**
To verify that all possible JSON output has been overridden, `Ctrl-F` for `def render` in `DeviseTokenAuth::RegistrationsController` and `DeviseTokenAuth::SessionsController`. Note that most of them call `render_error` which is overridden in `app/controllers/concerns/auth_helper.rb` as a concern since it is used across Devise controllers.

Also removed Axios interceptor for transforming snake case to camel case in frontend.

### CODE REVIEW

File: backend/app/controllers/concerns/auth_helper.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# frozen_string_literal: true
+
+module AuthHelper
+  def render_error(status, message, _data = nil)
```
Comment (ianyong) [inline backend/app/controllers/concerns/auth_helper.rb:4]: `data` is not needed because we want to use our own API response template.

**Comment by sebastiantoh** [comment]
1) Hmm, do you get this error when attempting to login / register in the frontend? 
`ActionView::MissingTemplate (Missing template auth/user with {:locale=>[:en], :formats=>[:html], :variants=>[], :handlers=>[:raw, :erb, :html, :builder, :ruby, :jbuilder]}. ...`

2) Does the devise controllers need to `include StatusMessages`? If i'm not wrong, the devise controllers do not inherit from our own application controller

3) Should the json keys in the rails response be in snake_case instead? This is so that the rails api is client-agnostic, and we leave the handling of the casing to the client

**Comment by ianyong** [comment]
> Weird, I got this once before but never managed to replicate it again. Now that I restarted my entire development environment, it seems to be back. Give me some time to figure this out, sorry.

If you go to `app/controllers/auth/sessions_controller.rb` and change line 11 from
```ruby
render 'auth/user', status: :created
```
to
```ruby
render json: 'auth/user', status: :created
```
and try to log in, you should get a different error. Then, when you revert it, logging in starts to work correctly even through server restarts (but not computer restarts). Weird.

**Comment by sebastiantoh** [comment]
> The JSON keys are in camel case not because of the client but because JSON is derived from JavaScript which typically uses camel case. I believe most of the popular JSON API style guides also recommend camel case for keys.

I see. From what I've read, there doesn't seem to be an actual naming standard. 

> To be safer, I've explicitly set the Devise parent controller in the Devise initialiser.

Oh nice, didn't realise you could explicitly set the parent controller.


thanks for helping to clean up the api! Lgtm!

**Pull request merged**